org.lflang/src/org/lflang/ASTUtils.java

```diff
@@ -1747,16 +1748,73 @@ public static String findAnnotationInComments(EObject object, String key) {
         return null;
     }
 
+    /**
+     * Return the value of the {@code @label} attribute if
+     * present, otherwise return null.
+     *
+     * @throws IllegalArgumentException If the node cannot have attributes
+     */
+    public static String findLabelAttribute(EObject node) {
+        List<Attribute> attrs = getAttributes(node);
+        return attrs.stream()
+                    .filter(it -> it.getAttrName().equals("label"))
+                    .map(it -> it.getAttrParms().get(0).getValue())
+                    .findFirst()
+                    .orElse(null);
+
+    }
+
+    /**
+     * Return the attributes declared on the given node. Throws
+     * if the node does not support declaring attributes.
+     *
+     * @throws IllegalArgumentException If the node cannot have attributes
+     */
+    public static List<Attribute> getAttributes(EObject node) {
+        if (node instanceof Reactor) {
+            return ((Reactor) node).getAttributes();
+        } else if (node instanceof Reaction) {
+            return ((Reaction) node).getAttributes();
+        } else if (node instanceof Action) {
+            return ((Action) node).getAttributes();
+        } else if (node instanceof Timer) {
+            return ((Timer) node).getAttributes();
+        } else if (node instanceof StateVar) {
+            return ((StateVar) node).getAttributes();
+        } else if (node instanceof Parameter) {
+            return ((Parameter) node).getAttributes();
+        } else if (node instanceof Input) {
+            return ((Input) node).getAttributes();
+        } else if (node instanceof Output) {
+            return ((Output) node).getAttributes();
+        }
+        throw new IllegalArgumentException("Not annotatable: " + node);
+    }
+
     /**
      * Search for an `@label` annotation for a given reaction.
-     * 
+     *
      * @param n the reaction for which the label should be searched
      * @return The annotated string if an `@label` annotation was found. `null` otherwise.
      */
     public static String label(Reaction n) {
```

Comment:
note that this overload could be removed if binary compatibility is not important, which seems likely.

Reply:
Sorry, I didn't get this comment. What binary compatibility?

Reply:
If someone has compiled a class that depends on this method and we remove it, their class will break. Minor releases generally preserve binary compatibility. I'm not sure whether this java API is publicly supported since you've started public releases... If the only ide extensions that use the java API are in this repo then we can most likely remove it (we're also pre-1.0.0)